File: persistence/src/main/java/io/sparkled/persistence/playlist/impl/query/GetAllPlaylistsQuery.java
package io.sparkled.persistence.playlist.impl.query;

import io.sparkled.model.entity.Playlist;
import io.sparkled.persistence.PersistenceQuery;
import io.sparkled.persistence.QueryFactory;

import java.util.List;

public class GetAllPlaylistsQuery implements PersistenceQuery<List<Playlist>> {

    @Override
    public List<Playlist> perform(QueryFactory queryFactory) {
        return queryFactory
                .selectFrom(qPlaylist)
                .orderBy(qPlaylist.name.asc())
                .fetch();
    }
}

File: persistence/src/main/kotlin/io/sparkled/persistence/playlist/impl/query/GetAllPlaylistsQuery.kt
package io.sparkled.persistence.playlist.impl.query

import io.sparkled.model.entity.Playlist
import io.sparkled.model.entity.QPlaylist.playlist
import io.sparkled.persistence.PersistenceQuery
import io.sparkled.persistence.QueryFactory

class GetAllPlaylistsQuery : PersistenceQuery<List<Playlist>> {

    override fun perform(queryFactory: QueryFactory): List<Playlist> {
        return queryFactory
            .selectFrom(playlist)
            .orderBy(playlist.name.asc())
            .fetch()
    }
}
